### PR TITLE
Update structure of job listings and add more roles

### PR DESCRIPTION
**Pull request by bradley-holt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,110 @@
...
+Read *Apply for a Summer 2021 IBM Quantum Internship* on the IBM Research Blog:  
+https://ibm.co/quantuminternships
 
-Search the [IBM jobs and careers website for more quantum computing opportunities](https://careers.ibm.com/ListJobs/All/Search/jobtitle/quantum/) at IBM.
+_For IBM Quantum internships in the United States, please apply by Monday, November 2, 2020 in order to receive the best chance at consideration._
```
Comment (hsaunders0788) [inline README.md:12]: Should we make a statement about international opportunities still being forthcoming?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,110 @@
...
 
-_Please apply by Monday, November 2, 2020 in order to receive the best chance at consideration._
+Applicants should submit their resumes to the appropriate job requisite(s) listed below. Cover letters and recommendation letters are _not_ required as part of your initial application.
```
Comment (hsaunders0788) [inline README.md]: I might take out "initial", we don't ask for a cover letter or recommendation letter at any point in the process as far as I'm aware?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,110 @@
...
+
+Graduate Quantum Developer Summer Intern (United States):  
+https://careers.ibm.com/ShowJob/Id/962202/Graduate-Quantum-Developer-Summer-Intern-2021/
+
+Undergraduate Quantum Developer Summer Intern (United States):  
+https://careers.ibm.com/ShowJob/Id/962204/Undergraduate-Quantum-Developer-Summer-Intern-2021/
+
+### Quantum Engineer Internships
+
+Engineering interns will have the opportunity to focus on hardware of software engineering. Areas of engineering could include superconducting qubits, quantum software architecture, cryoelectronic circuit design, quantum microwave engineering, and quantum FPGA engineering (among other areas).
```
Comment (hsaunders0788) [inline README.md]: hardware of software -> hardware or software

File: README.md [hunk 4]
```diff
@@ -1,18 +1,110 @@
...
+
+Locations:
+
+- IBM Thomas J. Watson Research Center (Yorktown Heights, New York, USA)
+- IBM Research — Almaden (San Jose, California, USA)
+
+IBM Quantum Offering Manager Intern (United States):  
+https://careers.ibm.com/ShowJob/Id/981532/IBM-Quantum-Offering-Manager-Intern-2021/
+
+### Quantum Community Advocate Internships
```
Comment (hsaunders0788) [inline README.md]: Should we move community advocate above designer and OM, given we want to hire more advocates?

File: README.md [hunk 5]
```diff
@@ -1,18 +1,110 @@
...
+
+Locations:
+
+- IBM Thomas J. Watson Research Center (Yorktown Heights, New York, USA)
+- IBM Research — Almaden (San Jose, California, USA)
+
+IBM Quantum Community Advocate Intern (United States):  
+https://careers.ibm.com/ShowJob/Id/980793/IBM%20Quantum%20Community%20Advocate%20Intern%20%20%202021
+
+## Early Professional Roles
```
Comment (hsaunders0788) [inline README.md]: I think this section would be clearer titled "Early Professional (Full Time) Roles"

File: README.md [hunk 6]
```diff
@@ -1,18 +1,110 @@
...
+
```
Comment (hsaunders0788) [inline README.md:110]: We have Quantum Specific Bulk Requisites that may be better to include than the wider research reqs. We don't have a separate postdoc one, but we match up fairly well with the others, and have a developer role also.

IBM Quantum Research Staff Member - 2021
https://careers.ibm.com/ShowJob/Id/986522/IBM-Quantum-Research-Staff-Member-2021/

IBM Quantum Entry Level Hardware Developer - 2021
https://careers.ibm.com/ShowJob/Id/986523/IBM-Quantum-Entry-Level-Hardware-Developer-2021/

IBM Quantum Entry Level Developer - 2021
https://careers.ibm.com/ShowJob/Id/981311/IBM-Quantum-Entry-Level-Developer-2021/

IBM Quantum Entry Level Engineer - 2021
https://careers.ibm.com/ShowJob/Id/980973/IBM-Quantum-Entry-Level-Engineer-2021/

**Review by hsaunders0788 — CHANGES_REQUESTED** [review]

**Review by hsaunders0788 — APPROVED** [review]

**Review by mmcelaney — APPROVED** [review]
Merging now!

**Pull request merged**